docs/chapter4-payload-handoff-format.rst

```diff
@@ -29,9 +29,834 @@ architecture, either 32 or 64 bits.
+
+.. tabularcolumns:: | p{3cm} p{0.75cm} p{2cm} p{9.5cm} |
+.. table:: /options/upl-image/image@<addr>
+
+   ================= =========== ================== ======================================================================================================================================================================
+   Property          Usage       Value Type         Definition
+   ================= =========== ================== ======================================================================================================================================================================
+   reg               R           prop-encoded array Address and size that the image was actually loaded to. This is normally the same as the image load address within the FIT, but can be different if anything was
+                                                    relocated, or if the FIT did not provide a load address.
+   offset            O           u32                Offset within FIT of the image node for this image.
+   description       R           string             Description value for this image from the FIT.
+   ================= =========== ================== ======================================================================================================================================================================
+
+
+4.2.5 Node: /pci (R)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Payload may need below FDT nodes for collecting PCI devices information on the
+system for binding corresponding drivers to provide services and functionality.
+
+There could be 2 or more root bridge nodes and even the same PCI segment may be
+shared by several root bridges. In this case ecam-base-addr in the PCI root
+bridge node is important for payload to know how to access each root bridge as
+well as what segment they belong to. (by masking bus/dev/func, the same
+ecam-base-addr will be the same segment root bridges).
+
+Under each root bridge node, there could be several root ports and/or PCI
+devices listed as its child nodes. Please refer the example use case below for
+reference implementation.
+
+The PCI bindings in this node must follow this specification - the IEEE Std
+1275-1994 Standard for Boot (Initialization Configuration) Firmware
+[PCIBusBinding]_ with some restrictions.
+
+As the #address-cells is always 3, the child / PCI address has to be presented
+in this format, **per PCI Bus Binding spec** [PCIBusBinding]_:
+
+* The non-prefetchable and prefetchable memory windows must each be exactly
+  256MB (0x10000000) in size.
+* The prefetchable memory window must be immediately adjacent to the
+  non-prefetcable memory window.
+
+Here is a bitmap of the 3-cell address which is followed by the explanation of
+each bit:
+
+* Cell 1: npt000ss bbbbbbbb dddddfff rrrrrrrr
+* Cell 2: hhhhhhhh hhhhhhhh hhhhhhhh hhhhhhhh
+* Cell 3: llllllll llllllll llllllll llllllll
+
+Notes:
+
+* n: relocatable region flag
```

Comment:
could we copy the pci bus binding spec which having each value meaning?
n is 0 if the address is relocatable, 1 otherwise
p is 1 if the addressable region is "prefetchable", 0 otherwise
t is 1 if the address is aliased (for non-relocatable I/O), below 1 MB (for Memory),
or below 64 KB (for relocatable I/O)

Reply:
Done.